### ISSUE TITLE
Tryit Code Action for the services are shown even when there are syntax errors

### ISSUE THREAD

**Issue by nadeeshaan** [wso2/ballerina-plugin-vscode#148]
**Description:**
Consider the following capture,

<img width="620" alt="Screenshot 2022-01-06 at 15 51 03" src="https://user-images.githubusercontent.com/1329674/148368168-054915de-6a0e-478c-b226-5e41e791f108.png">

Here the source is having a syntax error and even in this case, we show the try it code action. We should not encourage the user/ navigate the user to such actions if the result of the action is ending in an error scenario. 

Shall we allow this only in scenarios where semantic/ syntax errors are not available

**Comment by HirumalPriyashan** [comment]
After this https://github.com/wso2-enterprise/ballerina-plugin-vscode/pull/294 PR, run, debug and try it code lenses will be available for user. Since it is required that project is in already running condition, to use try out code action, any errors will be resulted at the execution time.

Additionally, if there is an already running service, user should be able to use the try it code action, (even though there are new changes which are causing semantic/ syntax errors) with the already running service.

**Comment by nadeeshaan** [comment]
@HirumalPriyashan I agree that we need a way to open the try it editor at any time. But, consider showing this for a newly added but incomplete source as shown below.

<img width="387" alt="Screenshot 2022-02-03 at 07 55 26" src="https://user-images.githubusercontent.com/1329674/152271203-57a85170-c643-431f-b32c-b0469c9ceebd.png">

Consider the following as well.

https://user-images.githubusercontent.com/1329674/152271808-7b3c31f0-a9d0-40ae-a66b-44242e6986a3.mp4

Both these are captured after installing v3.0.0-m1 (after merging PR mentioned)

In this particular scenario, when the user clicks on the code lens, we prompt with an error which interrupts the user's expectation. 

If we need to allow the user to open the try it editor even with errors in the source, we should find a separate way of enabling it without the on the fly validation.

**Comment by HirumalPriyashan** [comment]
@nadeeshaan 
Even though there is a running service after changing the source to have errors, the same error message in the second capture will prompt.
So its better to have a separate way to validate source before enabling code lenses (which may require a separate LS call) or inform the user with a proper error message.

**Comment by nadeeshaan** [comment]
> 

@HirumalPriyashan yes, that's my point as well. We should not show the code lens in erroneous syntax and the implementation detail is okay to decide whether we use LS calls or not
